src/librustc/hir/lowering.rs

```diff
@@ -5298,13 +5298,23 @@ impl<'a> LoweringContext<'a> {
 
     fn elided_path_lifetime(&mut self, span: Span) -> hir::Lifetime {
         match self.anonymous_lifetime_mode {
-            // N.B., We intentionally ignore the create-parameter mode here
-            // and instead "pass through" to resolve-lifetimes, which will then
-            // report an error. This is because we don't want to support
-            // impl elision for deprecated forms like
+            // In create-parameter mode we error here because we don't want to support
+            // deprecated impl elision in new features like impl elision and `async fn`,
+            // both of which work using the `CreateParameter` mode:
```

Comment:
I understand why this is happening but I find it rather strange that `async fn`s and `fn`s don't act similarly. That said, I agree with the consensus in the meeting that we can change our minds later if need be.